src/SymfonyCache/TagSubscriber.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use FOS\HttpCache\SymfonyCache\Tag\ManagerInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Adds tag invalidation capabilities to the Symfony HTTP cache
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ * 
+ * {@inheritdoc}
+ */
+class TagSubscriber implements EventSubscriberInterface
+{
+    /**
+     * HTTP method to use for remote tag invalidation.
+     */
+    const METHOD_INVALIDATE = 'INVALIDATE';
+    
+    /**
+     * Name for HTTP header containing the tags (for both invalidation and
+     * initial tagging).
+     */
+    const HEADER_TAGS = 'X-TaggedCache-Tags';
+
+    /**
+     * Header which should contain the content digest produced by the Symfony
+     * HTTP cache.
+     */
+    const HEADER_CONTENT_DIGEST = 'X-Content-Digest';
+
+    /**
+     * @var ManagerInterface
+     */
+    private $tagManager;
+
+    /**
+     * @param ManagerInterface $tagManager
+     * @param mixed $options
+     */
+    public function __construct(ManagerInterface $tagManager)
+    {
+        $this->tagManager = $tagManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::PRE_HANDLE => 'preHandle',
+            Events::POST_HANDLE => 'postHandle',
+        ];
+    }
+
+    /**
+     * Check to see if the request is an invalidation request, if so
+     * handle the invalidation.
+     *
+     * @param CacheEvent $event
+     */
+    public function preHandle(CacheEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (self::METHOD_INVALIDATE!== $request->getMethod()) {
```

Comment:
Formatting -- note that this method does not work for 2 reasons with the built-in PHP webserver:

1. It does not recognize custom HTTP methods.
2. It is single threaded, making a request on the webserver when it is serving will cause it to lockup.

Should document this.